### DISCUSSION TITLE
Sending email with lithium

### DISCUSSION THREAD

**Discussion by shivshankardayal** [matt-42/lithium#73] [Q&A]
What library will you recommend to send email with lithium?

**Comment by matt-42** [comment]
Hi @shivshankardayal , I never dealt with this problem before. But I would do it this way:
   - build a thread safe queue of outgoing mails.
   - in the server API handlers, push mails to the queue.
   - in parallel to the server, start a thread that uses a smtp client library (libcurl for example) to process the queue.

This page shows how to use libcurl to send mails: https://curl.se/libcurl/c/smtp-mail.html

**Reply by shivshankardayal** [reply]
Point no. 1 and 2 I understand. Can you provide code for step 3 please or a complete example because it is quite common for a web application to send email?

**Reply by matt-42** [reply]
I'll add this to my todolist. In the meantime you can check this libcurl example: https://curl.se/libcurl/c/smtp-mail.html

**Reply by shivshankardayal** [reply]
Thanks, I am aware of `libcurl` and its facilities.
